public/js/src/module/connection.js

```diff
@@ -609,34 +614,53 @@ function getMediaFile(url) {
 /**
  * Obtains a data/text file
  *
- * @param { string } url - URL to data tile
+ * @private
+ * @param {id} string
+ * @param {string} url - URL to data tile
  * @param {object } languageMap - language map object with language name properties and IANA subtag values
  * @return {Promise<XMLDocument>} a Promise that resolves with an XML Document
  */
-function getDataFile(url, languageMap) {
+function getDataFile(id, url, languageMap) {
     let contentType;
 
     return fetch(url)
         .then((response) => {
-            contentType = response.headers.get('Content-Type').split(';')[0];
+            contentType =
+                response.headers.get('Content-Type')?.split(';')[0] ?? '';
+
+            if (
+                (contentType === '' || String(contentType) === 'null') &&
```

Comment:
Note for the future: This may look a little bit wacky. It's like this because geojson is a newer content type and because it's a semantic layer over json. Our spec requires a geojson extension so that's what this looks for.